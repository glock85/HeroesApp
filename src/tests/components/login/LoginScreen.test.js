import { MemoryRouter, Route, Routes } from "react-router-dom"
import { mount } from 'enzyme'
import { AuthContext } from "../../../auth/authContext"
import { types } from "../../../types/types";
import LoginScreen from "../../../components/login/LoginScreen";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('Pruebas en <Login / >', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={ contextValue }>
            <MemoryRouter initialEntries={ ['/login'] }>
                <Routes>
                    <Route path="/login" element={ <LoginScreen /> } />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>
    )

    test('debe de hacer match con el snapshot', () => {

        expect( wrapper ).toMatchSnapshot();

    })

    test('debe de hacer dispatch y la navegacion', () => {

        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();

        expect( contextValue.dispatch ).toHaveBeenCalledWith({
            type: types.login,
            payload: { name: 'joseph'}
        });

        expect( mockNavigate ).toHaveBeenCalledWith( '/marvel', { replace: true } );

        localStorage.setItem( 'lastPath', '/dc' );

        handleClick();

        expect( mockNavigate ).toHaveBeenCalledWith( '/dc', { replace: true } );
        
    })

})