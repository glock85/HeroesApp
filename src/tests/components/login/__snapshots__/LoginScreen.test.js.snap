// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Login / > debe de hacer match con el snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-dark"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
